include/rocksdb/slice.h

```diff
@@ -213,9 +213,19 @@ inline bool operator!=(const Slice& x, const Slice& y) {
   return !(x == y);
 }
 
+// UBSAN complain that we pass nullptr to memcmp that's fine since
+// we always do that for a string of len = 0
+#ifdef ROCKSDB_UBSAN_RUN
+#if defined(__clang__)
+__attribute__((__no_sanitize__("undefined")))
+#elif defined(__GNUC__)
+__attribute__((__no_sanitize_undefined__))
+#endif
+#endif
 inline int Slice::compare(const Slice& b) const {
   const size_t min_len = (size_ < b.size_) ? size_ : b.size_;
   int r = memcmp(data_, b.data_, min_len);
```

Comment:
I still feel we should fix it since this is undefined behavior means return value can be anything.